ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto

```diff
@@ -50,14 +52,30 @@ message CreatedEvent {
   // Required
   Identifier template_id = 3;
 
-  // The key of the created contract, if defined.
+  // The key of the created contract.
+  // This will be set if and only if ``create_arguments`` is set and ``template_id`` defines a contract key.
   // Optional
   Value contract_key = 7;
 
   // The arguments that have been used to create the contract.
-  // Required
+  // Set either:
+  //   - if there was a party, which is in the ``witness_parties`` of this event,
+  //     and for which an ``InclusiveFilters`` exists with the ``template_id`` of this event
+  //     among the ``template_ids``,
+  //   - or if there was a party, which is int the ``witness_parties`` of this event,
+  //     and for which a wildcard filter exists (``Filters`` without ``InclusiveFilters``,
+  //     or with an ``InclusiveFilters`` with empty ``template_ids`` and empty ``interface_filters``).
```

Comment:
so `interface_filters` **empty** and `template_ids` **empty** is a templateId wildcard but if `interface_filters` **non-empty**, `template_ids` empty is not treated as a wildcard anymore. If this irregularity in the behavior of `template_ids` is intended, can someone explain why?

Reply:
I think this is mainly backwards compat:

If you don’t set template ids and don’t set interface ids we have to keep the existing behavior of it being a wildcard.

On the other hand, we do want to support you only subscribing to an interface and there is no way to say "don't subscribe to any templates" so we use the presence of the interface filter to turn off the template wildcard behavior.

Reply:
So hybrid subscriptions (interface id and unrelated template id) won't be supported.

Reply:
I see. So when turning on `interface_filters` we are narrowing down the fetch scope for a party and since wildcard `template_ids` projection fully includes any projection due to `interface_filters`, we are fine with this behavior

Reply:
> since wildcard template_ids projection fully includes any projection due to interface_filters,

I'm not quite sure I fully understand what you're saying here. A wildcard template id query does not include interface projections. But it includes the events for which you may want to have interface projections.

Put differently: Interface subscriptions are for when you don’t want to or cannot deal with specific templates and want an abstraction over a set of potentially unknown (to the caller) templates. A wildcard template filter is basicallly saying "I can handle any template" so at that point you don’t need interface filters.